string.c

```diff
@@ -3418,13 +3419,34 @@ str_casecmp_p(VALUE str1, VALUE str2)
     return rb_str_eql(folded_str1, folded_str2);
 }
 
+static long
+strseq_core(const char *str_ptr, const char *str_ptr_end, long str_len,
```

Comment:
Yes, this function is extracted from `rb_strseq_index` because I needed to use this from `undump_after_backslash` too.